app/models/daos/slick/UserDAOSlick.scala

```diff
@@ -191,6 +192,94 @@ object UserDAOSlick {
     anonUsers ++ otherUsers
   }
 
+  /**
+   * Returns a count of all users who have ever started (or completed) an audit or validation task (across all roles).
+   * That is, this method counts the number of unique users across both audit users and validation users
+   *
+   * This method is deprecated. Use the parameterized version of countAllUsersContributed instead
+   *
+   * @param taskCompleted
+   * @return
+   */
+//  def countAllUsersContributedOld(taskCompleted: Boolean): Int = db.withSession { implicit session =>
+//    val tasks = if (taskCompleted) AuditTaskTable.completedTasks else auditTaskTable
+//    val users = tasks.map(_.userId) ++ LabelValidationTable.validationLabels.map(_.userId)
+//
+//    // groupBy gives us a list of pairs of userid to userid and then map gives us
+//    // the first element of the pair for each item
+//    val distinctUsers = users.groupBy(x => x).map(_._1)
+//
+//    distinctUsers.size.run
+//  }
+
+
+  /**
+   * Returns a count of all users who completed at least one audit task or at least one validation given
+   * the following parameters
+   *
+   * @param timeInterval: can be "today", "yesterday", "week", or "month". If anything else, defaults to "all time"
+   * @param taskCompletedOnly: if true, only counts users who have completed one audit task or at least one validation.
+   *                           Defaults to false.
+   * @param highQualityOnly: if true, only counts users who are marked as high quality. Defaults to false.
+   * @return
+   */
+  def countAllUsersContributed(timeInterval: String = "all time", taskCompletedOnly: Boolean = false, highQualityOnly: Boolean = false): Int = db.withSession { implicit session =>
+
+    // Build up SQL string related to validation and audit task time intervals
+    // Defaults to *not* specifying a time (which is the same thing as "all time")
```

Comment:
Not doing a full code review rn, but I looked over this code earlier to help me write some code for the leaderboard. I rewrote part of it in a way that is more Scala-y (notably, no mutable variables):

```
val lblValidationTimeIntervalSql, auditTaskTimeIntervalSql = timeInterval.toLowerCase() match {
    case "today" => (
        """WHERE (label_validation.end_timestamp AT TIME ZONE 'PST')::date = (NOW() AT TIME ZONE 'PST')::date""",
        """AND (audit_task.task_end AT TIME ZONE 'PST')::date = (NOW() AT TIME ZONE 'PST')::date"""
    )
    case "yesterday" => (
        """WHERE (label_validation.end_timestamp AT TIME ZONE 'PST')::date = (now() AT TIME ZONE 'PST')::date - interval '1' day""",
        """AND (audit_task.task_end AT TIME ZONE 'PST')::date = (now() AT TIME ZONE 'PST')::date - interval '1' day"""
    )
    case ...
}
```

Reply:
I'm doing something very similar for the leaderboard: https://github.com/ProjectSidewalk/SidewalkWebpage/blob/2177-gamification-leaderboard/app/models/user/UserStatTable.scala#L146